### PR TITLE
[P1][Beta] Fix phazing moves forcing switches into fainted/ineligible Pokemon

### PR DESCRIPTION
**Pull request by PigeonBar**
## What are the changes the user will see?

Phazing moves can no longer randomly choose to force a switch into a fainted/ineligible Pokemon.
Also, phazing moves properly use seeded RNG instead of unseeded RNG.

## Why am I making these changes?

Bug report: [Discord](https://discord.com/channels/1125469663833370665/1176874654015684739/1312149844797685820)

## What are the changes from a developer perspective?

1. Phazing moves now generate a list of indices of eligible Pokemon to switch into, and then randomly pick an eligible index using seeded RNG.
2. Added tests for Whirlwind and Dragon Tail.

## Screenshots/Videos

<details>
<summary> Before the changes: Phazing moves can force a switch into a fainted Pokemon </summary>

https://github.com/user-attachments/assets/400bf42b-657b-44fc-ad84-ac7f7a96825f

</details>
<details>
<summary> After the changes: Phazing moves no longer force a switch into a fainted Pokemon </summary>

https://github.com/user-attachments/assets/18905718-2be9-49cc-b9d5-2ca9681ec77b

</details>

## How to test the changes?

1. Reach the floor 195 rival.
2. Defeat 4 of the rival's 6 Pokemon.
3. Use a phazing move.
4. The phazing move should always force the rival to switch to their only other unfainted Pokemon.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes manually?
- [x] Are all unit tests still passing? (`npm run test`)
  - [x] Have I created new automated tests (`npm run create-test`) or updated existing tests related to the PR's changes?
- [x] Have I provided screenshots/videos of the changes (if applicable)?
  - ~~[ ] Have I made sure that any UI change works for both UI themes (default and legacy)?~~

Are there any localization additions or changes? If so:
- ~~[ ] Has a locales PR been created on the [locales](https://github.com/pagefaultgames/pokerogue-locales) repo?~~
  - ~~[ ] If so, please leave a link to it here:~~
- ~~[ ] Has the translation team been contacted for proofreading/translation?~~

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -5975,14 +5975,22 @@ export class ForceSwitchOutAttr extends MoveEffectAttr {
         }
       }
 
-      if (switchOutTarget.scene.getPlayerParty().filter((p) => p.isAllowedInBattle() && !p.isOnField()).length < 1) {
+      // Find indices of off-field Pokemon that are eligible to be switched into
+      const eligibleNewIndices = [] as number[];
```
Comment (DayKev) [inline src/data/move.ts:5979]:
```suggestion
      const eligibleNewIndices: number[] = [];
```

File: src/data/move.ts [hunk 2]
```diff
@@ -6011,14 +6019,22 @@ export class ForceSwitchOutAttr extends MoveEffectAttr {
       }
       return false;
     } else if (user.scene.currentBattle.battleType !== BattleType.WILD) { // Switch out logic for enemy trainers
-      if (switchOutTarget.scene.getEnemyParty().filter((p) => p.isAllowedInBattle() && !p.isOnField()).length < 1) {
+      // Find indices of off-field Pokemon that are eligible to be switched into
+      const eligibleNewIndices = [] as number[];
```
Comment (DayKev) [inline src/data/move.ts:6023]:
```suggestion
      const eligibleNewIndices: number[] = [];
```

**Review by DayKev — COMMENTED** [review]
Just a minor style nit but otherwise seems good.

**Review by DayKev — APPROVED** [review]

**Review by torranx — APPROVED** [review]
lgtm 🚀

**Pull request merged**